### PR TITLE
Api 726/integrations detailed view

### PR DESCRIPTION
**Pull request by scefali**
This PR creates a base class for the integration detailed view called `AbstractIntegrationDetailedView` that each of the detailed views inherits from. The purpose of this PR is to reduce the amount of duplicated code as now much of the view logic sits in the abstract base class. This will also make it much easier to add document-based integrations at a later date and make the format align with the rest of the integrations.

The views should be exactly the same except some changes to the Sentry App detailed view so it aligns better with first-party integrations and plugins.

Before:
Screen Shot 2020-02-13 at 3 42 18 PM

After:

Screen Shot 2020-02-13 at 3 42 06 PM

Not a big deal since we are still waiting for designs and much will change.

Still need to add some docstrings to the methods in the base class.

### CODE REVIEW

File: src/sentry/static/sentry/app/types/index.tsx [hunk 1]
```diff
@@ -1,6 +1,16 @@
 import {SpanEntry} from 'app/components/events/interfaces/spans/types';
 import {API_ACCESS_SCOPES} from 'app/constants';
 import {Field} from 'app/views/settings/components/forms/type';
+import {
+  INSTALLED,
+  NOT_INSTALLED,
+  PENDING,
+} from 'app/views/organizationIntegrations/constants';
+
+export type IntegrationInstallationStatus =
```
Comment (scefali) [inline src/sentry/static/sentry/app/types/index.tsx]: Finally made a type for the installation status :)

File: src/sentry/static/sentry/app/views/organizationIntegrations/constants.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
-export const INSTALLED = 'Installed';
-export const NOT_INSTALLED = 'Not Installed';
-export const PENDING = 'Pending';
+export const INSTALLED = 'Installed' as const;
+export const NOT_INSTALLED = 'Not Installed' as const;
+export const PENDING = 'Pending' as const;
```
Comment (NisanthanNanthakumar) [inline src/sentry/static/sentry/app/views/organizationIntegrations/constants.tsx]: @scefali Do we need type casting here?

File: src/sentry/static/sentry/app/types/index.tsx (hunk 1 above)
Comment (NisanthanNanthakumar) [reply src/sentry/static/sentry/app/types/index.tsx]: @scefali thanks for updating the other components with the new type!

File: src/sentry/static/sentry/app/views/organizationIntegrations/constants.tsx (hunk 2 above)
Comment (scefali) [reply src/sentry/static/sentry/app/views/organizationIntegrations/constants.tsx]: @NisanthanNanthakumar This isn't typecasting. This is called a const assertion (https://mariusschulz.com/blog/const-assertions-in-literal-expressions-in-typescript). This is so I can use the constants as both a type and a value.

**Review by NisanthanNanthakumar — APPROVED** [review]

**Pull request merged**